client/my-sites/checkout/upsell-nudge/professional-email-upsell/index.tsx

```diff
@@ -39,21 +46,29 @@ type ProfessionalEmailUpsellProps = {
 	domainName: string;
 	handleClickAccept: ( action: string ) => void;
 	handleClickDecline: () => void;
-	productCost?: number | null;
-	productSlug: string;
 	setCartItem: ( cartItem: MinimalRequestCartProduct, callback: () => void ) => void;
+	intervalLength?: IntervalLength | undefined;
 };
 
 const ProfessionalEmailUpsell = ( {
 	currencyCode,
 	domainName,
 	handleClickAccept,
 	handleClickDecline,
-	productCost,
-	productSlug,
 	setCartItem,
+	intervalLength = IntervalLength.ANNUALLY,
 }: ProfessionalEmailUpsellProps ) => {
 	const translate = useTranslate();
+	const dispatch = useDispatch();
+	const [ selectedIntervalLength, setSelectedIntervalLength ] = useState( intervalLength );
+	const productCost = useSelector( ( state ) => {
+		if ( selectedIntervalLength === IntervalLength.ANNUALLY ) {
+			return getProductCost( state, TITAN_MAIL_YEARLY_SLUG );
+		}
+		return getProductCost( state, TITAN_MAIL_MONTHLY_SLUG );
```

Comment:
Nit: for clarity, it may be worth adding a newline here to make it extra clear that the following code is inside the `if` block.
```suggestion

		return getProductCost( state, TITAN_MAIL_MONTHLY_SLUG );
```